Tensile/KernelWriterAssembly.py

```diff
@@ -7611,56 +7815,92 @@ def localReadDo(self, kernel, bufferIdx, iui, epsi, uIdx, tP):
       pack = Code.Module("pack%s"%tc)
       isTailRemainderLoop = self.getTmpSgpr(2)
       tmpVgprIdx = self.vgprPool.checkOut(self.numVgprValuAPerBlock if tc == 'A' else self.numVgprValuBPerBlock)
       packCode = pack.addCode (Code.Module("packCode"))
       packCode.addTempVgpr(tmpVgprIdx)
       for vIdx in range(0, numVectorsPerTile):
-        for rIdx in range(0, numReadsPerVector):
           # emit an instruction for each half-dword load along k due to strided access
           # checkout 2 vregs for each half-dword loads
-          for kIdx in range(0, numReadsAlongK):
+        for rIdx in range(0, numReadsPerVector):
+          for oIdx in range(0, numOffsets):
             localReadCode = imod.addCode (Code.Module("LocalRead%s Valu%u"%(tc,valuIdx)))
             paramList = []
             destVgpr = vgpr("Valu%s_X%u_I%u+%u" % (tc, bufferIdx, iui, int(valuIdx))) # inc by 1 every 2 kIdx
             tmpVgpr = vgpr(tmpVgprIdx+int(valuIdx)) # inc by 1 every 2 kIdx
             tIdx = tP["tensorIdx"]
-            if kIdx%2 == 0:
+            if rIdx%2 == 0:
               paramList.append(destVgpr)
             else:
               paramList.append(tmpVgpr)
             paramList.append(vgpr("LocalReadAddr%s"%tc))
-            offset = ((rIdx * kernel["MacroTile%u" % tIdx] // kernel["ThreadTile%u" % tIdx] + kIdx*(kernel["MacroTile%s"%tc]+kernel["LdsPad%s"%tc]) + tP["localReadOffset"]) \
-                 *tP["bpe"]+tP["localReadSwapByteOffset"])//offsetMultiplier
-            paramList.append(int(offset))
-            oIdx = 0
+            if tP["isA"]:
+              if kernel["MatrixInstB"] == 1:
+                outputBlocks = 1 
+              elif kernel["MIWG0"] > globalParameters["WavefrontWidth"] :
+                outputBlocks = globalParameters["WavefrontWidth"]//kernel["MatrixInstM"]
+              else:
+                outputBlocks = kernel["MIWG0"] // kernel["MatrixInstM"]
+              offset = ((((rIdx * kernel["MacroTile%u" % tIdx] + kernel["LdsPad%s"%tc]) + outputBlocks *(vIdx*numOffsets+oIdx)*kernel["MatrixInstM"] + \
+                + tP["localReadOffset"])*tP["bpe"]+tP["localReadSwapByteOffset"])//offsetMultiplier)
+              paramList.append(int(offset))
+            if tP["isB"]:
+              if kernel["MatrixInstB"] == 1 or  kernel["MIWG0"] > globalParameters["WavefrontWidth"] :
+                outputBlocks = 1
+              else:
+                outputBlocks = globalParameters["WavefrontWidth"] // (kernel["MIWG0"])
+                #TODO: colBlocks are  non-contiguous col block0 from simd0,simd1,simd2, simd col block1 from simd0,simd1,...
+                  # StoreoffsetCalculation requireFix
+               
+              offset = ((((rIdx * kernel["MacroTile%u" % tIdx] + kernel["LdsPad%s"%tc]) + (outputBlocks*(vIdx*numOffsets+oIdx)*kernel["MatrixInstN"]) \
```

Comment:
Duplicate of above

Reply:
localReadDo requires one more round simplification.. may be next version